### PR TITLE
FF94 : Updates to sanitizer API

### PR DESCRIPTION
**Pull request by hamishwillee**
This is part of update for FF94 in  #9366

The API has the following changes w.r.t. current docs (by inspection and testing)
- [x] `Sanitizer.getConfiguration()` - not going to add because it is not implemented
- [x] `Sanitizer.sanitizeToString()` method - remove doc - never implemented.
- [x] `Sanitizer.sanitizeFor()` method - add new doc
- [x]  `Element.setHTML()` method - add new doc
- [x] Tidy up configuration options - new but not actually supported.
- [x] Add `Sanitizer.sanitizeFor()` to `Sanitizer`
- [x] Update the top level API doc with all of it.

### CODE REVIEW

File: files/en-us/web/api/element/index.md [hunk 1]
```diff
@@ -87,6 +87,8 @@ _`Element` inherits properties from its parent interface, {{DOMxRef("Node")}}, a
   - : Returns the open shadow root that is hosted by the element, or null if no open shadow root is present.
 - {{DOMxRef("Element.openOrClosedShadowRoot")}} {{Non-standard_Inline}}{{readOnlyInline}}
   - : Returns the shadow root that is hosted by the element, regardless if its open or closed. **Available only to [WebExtensions](/en-US/docs/Mozilla/Add-ons/WebExtensions).**
+- {{DOMxRef("Element.setHTML")}}
+  - : Applies a string using a {{DOMxRef("Sanitizer")}} directly to an existing element node.
```
Comment (hamishwillee) [inline files/en-us/web/api/element/index.md:91]: Will improve this text when I understand what it means.

File: files/en-us/web/api/html_sanitizer_api/index.md [hunk 2]
```diff
@@ -60,7 +56,7 @@ const result = new Sanitizer().sanitize(stringToClean);
     </tr>
     <tr>
       <td>
-        {{SpecName('HTML Sanitizer API','#sanitizer-api','sanitizeToString')}}
+        {{SpecName('HTML Sanitizer API','#sanitizer-api')}}
```
Comment (hamishwillee) [inline files/en-us/web/api/html_sanitizer_api/index.md:59]: How do we currently do this for APIs - I can add the main classes but they don't link to the top of the spec as this link would.

File: files/en-us/web/api/html_sanitizer_api/index.md [hunk 3]
```diff
@@ -7,39 +7,35 @@ tags:
...
-// our input string to clean
-const stringToClean = 'Some text <b><i>with</i></b> <blink>tags</blink>, including a rogue script <script>alert(1)</script> def.';
-
-const result = new Sanitizer().sanitizeToString(stringToClean);
-console.log(result);
-// Logs: "Some text <b><i>with</i></b>, including a rogue script def."
-```
-
-The other method available is the {{domxref('Sanitizer.sanitize()')}} method. Which is very similar to above, however returns a {{domxref('DocumentFragment')}} with disallowed `script` and `blink` elements removed.
+This example shows the use of the {{domxref('Sanitizer.sanitize()')}} method.
```
Comment (hamishwillee) [inline files/en-us/web/api/html_sanitizer_api/index.md:37]: Need to add example of `Element.setHTML` too. So far just removed the invalid `sanitizeToString()` example

File: files/en-us/web/api/sanitizer/sanitizer/index.md [hunk 4]
```diff
@@ -2,53 +2,52 @@
...
+    - `allowAttributes` {{optional_inline}}
+      - : An {{jsxref('Array')}} of {{jsxref('String','strings')}} indicating attributes the sanitizer should not remove.
+    - `dropAttributes` {{optional_inline}}
+      - : An {{jsxref('Array')}} of {{jsxref('String','strings')}} indicating attributes the sanitizer should remove.
+    - `allowCustomElements` {{optional_inline}}
+      - : `false` (default) Drops custom elements and their children.
+        `true` ensures that custom elements are checked against build-in and custom configuration checks.
+    - `allowComments` {{optional_inline}}
+      - : `false` (default) Drops HTML comments.
+        `true` ensures that comments are retained.
```
Comment (hamishwillee) [inline files/en-us/web/api/sanitizer/sanitizer/index.md:43]: Confirming this with spec (i.e. that no actual sanitization can ever be needed of the comment text)

File: files/en-us/web/api/sanitizer/sanitizer/index.md [hunk 5]
```diff
@@ -2,53 +2,52 @@
...
+> Due to this the parameter has not been implemented.
 
 ## Examples
 
-This example shows the result of sanitizing a string with disallowed
-`script` elements.
+This example shows the result of sanitizing a string with disallowed `script` elements.
 
 ```js
 new Sanitizer().sanitizeToString("abc <script>alert(1)</script> def");
```
Comment (hamishwillee) [inline files/en-us/web/api/sanitizer/sanitizer/index.md:53]: Will need to remove this example as it uses removed string: `sanitizeToString()` example

**Comment by hamishwillee** [comment]
Can I please get a review on this next week?  Good to not have these things hanging on too long.

FYI @Rumyra this isn't urgent because the changes are behind a preference (i.e. not "shipped" in FF94). The API is still in flux - what this documents is the things that have been implemented - there are some bits still to come.

**Review by Rumyra — CHANGES_REQUESTED** [review]
Only a couple of minor things - thanks @hamishwillee this is awesome

**Review by Rumyra — APPROVED** [review]

**Pull request merged**